src/App.js

```diff
@@ -2,18 +2,110 @@ import React from 'react';
 import './App.css';
 
 import users from './api/users';
+import todos from './api/todos';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+import { TodoList } from './components/TodoList';
+
+class App extends React.Component {
+  state = {
+    userId: '',
+    title: '',
+    titleError: false,
+    userError: false,
+    todosToRender: [...todos],
+  };
+
+  handleChange = (event) => {
+    const { name, value } = event.target;
+
+    this.setState({
+      [name]: value.replace(/\ws+/g, '').trim(),
+      userError: false,
+      titleError: false,
+    });
+  }
+
+  formsCheck = () => {
+    const { userId, title, todosToRender } = this.state;
+
+    if (userId && title) {
+      todosToRender.push({
```

Comment:
it's still here

Reply:
But now it`s part of the state and it doesn`t mutate data

Reply:
that's why you need setState, you can't do this.state.something.push...